### PR TITLE
Telegraf kvm

### PR DESCRIPTION
**Pull request by vhe182**
The goal of this PR is to add support for kvm dashboards in grafana by adding the exec plugin.  
@meteorfox @besanradwan

### CODE REVIEW

File: roles/misc-perftools/tasks/main.yml [hunk 1]
```diff
@@ -13,6 +13,20 @@
     - iotop
     - bwm-ng
 
+- name: Create directory for kvm metric collection configuration file
+  file: path=/opt/telegraf state=directory mode=0755
+  when: "'compute_hosts' in group_names"
+
+- name: Set configuration file for kvm metric collection
+  template:
+    src: /root/os-ansible-perftools/roles/misc-perftools/templates/kvm_virsh.py
```
Comment (meteorfox) [inline roles/misc-perftools/tasks/main.yml]: This won't work. Don't use the template module, there's a file module. You need to put the file under the `files/` directory for this role, and just use the basename `kvm_virsh.py`. 

I'm not sure if you have to prefix the name with 'files/' or not. Here's Kevin's example.
https://github.com/openstack/openstack-ansible-ops/blob/2a1e702209cc713861bae54be2324cd2a59b3121/cluster_metrics/playbook-influx-telegraf.yml#L38-L44

File: roles/misc-perftools/templates/kvm_virsh.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python
```
Comment (meteorfox) [inline roles/misc-perftools/templates/kvm_virsh.py]: Please add comments indicating that this is not ours, and give credits to the author(s). If their project had a license, add the license as comment

File: group_vars/hosts/telegraf.yml [hunk 3]
```diff
@@ -48,3 +48,4 @@ telegraf_plugins:
       interfaces:
         - "bond0"
         - "bond1"
+  - name: exec
```
Comment (meteorfox) [inline group_vars/hosts/telegraf.yml:51]: Where are the missing options to tell exec what to execute?

https://github.com/openstack/openstack-ansible-ops/blob/2a1e702209cc713861bae54be2324cd2a59b3121/cluster_metrics/templates/telegraf.conf.j2#L62-L65

**Review by meteorfox — CHANGES_REQUESTED** [review]

File: roles/misc-perftools/templates/kvm_virsh.py (hunk 2 above)
Comment (vhe182) [reply roles/misc-perftools/templates/kvm_virsh.py]: Added the license file and included the short version of the Apache 2.0 license to the head of kvm_virsh.py

File: group_vars/hosts/telegraf.yml (hunk 3 above)
Comment (vhe182) [reply group_vars/hosts/telegraf.yml:51]: Thanks for catching this and clearing up my misunderstanding.  I have added the options and have tested them.

File: roles/misc-perftools/tasks/main.yml (hunk 1 above)
Comment (vhe182) [reply roles/misc-perftools/tasks/main.yml]: I used the copy module and it works as expected.  The example you reference uses the copy module.

**Comment by vhe182** [comment]
Thanks for the feedback.  I have made all changes and the PR is ready for re-review.

File: roles/misc-perftools/tasks/main.yml [hunk 4]
```diff
@@ -13,6 +13,19 @@
...
 
+- name: Create directory for kvm metric collection configuration file
+  file: path=/opt/telegraf state=directory mode=0755
+  when: "'compute_hosts' in group_names"
+
+- name: Set configuration file for kvm metric collection
+  copy:
+    src: "files/{{ item }}"
+    dest: "/opt/telegraf/{{ item }}"
+    mode: 0755
```
Comment (meteorfox) [inline roles/misc-perftools/tasks/main.yml:26]: Add `owner` and `group` as `'telegraf'`

**Review by meteorfox — CHANGES_REQUESTED** [review]
LGTM.

Just add that minor change for `owner` and `group`.

File: roles/misc-perftools/tasks/main.yml (hunk 4 above)
Comment (vhe182) [reply roles/misc-perftools/tasks/main.yml:26]: okey doke.

**Review by meteorfox — APPROVED** [review]
Thanks! LGTM

**Pull request merged**